bundles/org.openhab.voice.voskstt/src/main/java/org/openhab/voice/voskstt/internal/VoskSTTService.java

```diff
@@ -0,0 +1,291 @@
+        + " Speech-to-Text", description_uri = SERVICE_CATEGORY + ":" + SERVICE_ID)
+public class VoskSTTService implements STTService {
+    private static final String VOSK_FOLDER = Path.of(OpenHAB.getUserDataFolder(), "vosk").toString();
+    private static final String MODEL_PATH = Path.of(VOSK_FOLDER, "model").toString();
+    static {
+        Logger logger = LoggerFactory.getLogger(VoskSTTService.class);
+        File directory = new File(VOSK_FOLDER);
+        if (!directory.exists()) {
+            if (directory.mkdir()) {
+                logger.info("vosk dir created {}", VOSK_FOLDER);
+            }
+        }
+    }
+    private final Logger logger = LoggerFactory.getLogger(VoskSTTService.class);
+    private final ScheduledExecutorService executor = ThreadPoolManager.getScheduledPool("OH-voice-voskstt");
+    private final LocaleService localeService;
+    private VoskSTTConfiguration config = new VoskSTTConfiguration();
+    private @Nullable Model model;
+
+    @Activate
+    public VoskSTTService(@Reference LocaleService localeService) {
+        this.localeService = localeService;
+    }
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        configChange(config);
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        configChange(config);
+    }
+
+    @Deactivate
+    protected void deactivate(Map<String, Object> config) {
+        try {
+            unloadModel();
+        } catch (IOException e) {
+            logger.warn("IOException unloading model: {}", e.getMessage());
+        }
+    }
+
+    private void configChange(Map<String, Object> config) {
+        this.config = new Configuration(config).as(VoskSTTConfiguration.class);
+        if (this.config.preloadModel) {
+            try {
+                loadModel();
+            } catch (IOException e) {
+                logger.warn("IOException loading model: {}", e.getMessage());
+            }
+        } else {
+            try {
+                unloadModel();
+            } catch (IOException e) {
+                logger.warn("IOException unloading model: {}", e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public String getId() {
+        return SERVICE_ID;
+    }
+
+    @Override
+    public String getLabel(@Nullable Locale locale) {
+        return SERVICE_NAME;
+    }
+
+    @Override
+    public Set<Locale> getSupportedLocales() {
+        return Set.of(localeService.getLocale(null));
+    }
+
+    @Override
+    public Set<AudioFormat> getSupportedFormats() {
+        return Set.of(
+                new AudioFormat(AudioFormat.CONTAINER_WAVE, AudioFormat.CODEC_PCM_SIGNED, false, null, null, 16000L));
+    }
+
+    @Override
+    public STTServiceHandle recognize(STTListener sttListener, AudioStream audioStream, Locale locale, Set<String> set)
+            throws STTException {
+        Future<?> task;
```

Comment:
Please suppress this variable which is now unused.
And take a look on SAT checks to be sure that this is clean.

Reply:
Done.
This is reported for all this kind of services 'Unused configuration reference with uri - voice:voskstt'. As it was already happening on others previous add-ons like the GoogleTSS I assumed it was a known error. Just in case you were not aware of it.